src/main/java/org/opensearch/neuralsearch/processor/combination/ScoreCombiner.java

```diff
@@ -106,7 +108,8 @@ private void combineShardScores(
             combinedNormalizedScoresByDocId,
             sortedDocsIds,
             getDocIdSortFieldsMap(compoundQueryTopDocs, combinedNormalizedScoresByDocId, sort),
```

Comment:
We already have a check isSortByScore in getDocIDSortFieldsMap method. Can we pull that and put it in line 103 so that it can be reused in https://github.com/opensearch-project/neural-search/pull/1277/files#diff-438c860966e2a639e10a3e503c8292f0bdcb31c385d5c206785343a946cbda73R262

Reply:
It will be a optimized way.

Reply:
in `getDocIdSortFieldsMap`. And passing `sort` object helped us to retrieve 2 values already `isSortEnabled` and `isSortByScore` 

we also have an extra check for 
https://github.com/opensearch-project/neural-search/blob/c3aac712538ac4a6ed72a46f899a195989d02d42/src/main/java/org/opensearch/neuralsearch/processor/combination/ScoreCombiner.java#L159